redaxo/src/addons/mediapool/pages/media.detail.php

```diff
@@ -65,7 +67,12 @@
         if (1 != $gf->getRows()) {
             $error = rex_i18n::msg('pool_file_not_found');
             $file_id = 0;
-        } elseif (!rex::getUser()->getComplexPerm('media')->hasCategoryPerm($gf->getValue('category_id')) || !rex::getUser()->getComplexPerm('media')->hasCategoryPerm($rex_file_category)) {
+        } elseif (
+            (!rex::getUser()->getComplexPerm('media')->hasCategoryPerm($gf->getValue('category_id'))
```

Comment:
Ich sehe an so vielen stellen so rießige rechte checks ...das macht es total unübersichtlich und schwierig nachzuvollziehen.. können wir die rausziehen in eine eigene klasse ? Mit leserlichen methodennamen?

Ggf. Wäre es alternativ gut, die ganzen rechte abfragen, wie sie aktuell noch im master branch sind auf eine klasse mit leserlichen methoden zurückzuführen, dass dieser PR hier ggf. Davon profitiert?

Reply:
@staabm @gharlan wie gehen wir damit nun um? Der Medienpool war und ist natürlich ein großes gebastelt und mit dem Feature hab ich mich dem Stil angepasst.

Reply:
Mein vorschlag hab ich ja oben gebracht 

> Ggf. Wäre es alternativ gut, die ganzen rechte abfragen, wie sie aktuell noch im master branch sind auf eine klasse mit leserlichen methoden zurückzuführen, dass dieser PR hier ggf. Davon profitiert (separater PR)?

Alles andere was diesen PR hier leserlicher/kleiner macht ist ebenfalls gewünscht (separate nachvollziehbare PRs)

> Der Medienpool war und ist natürlich ein großes gebastelt und mit dem Feature hab ich mich dem Stil angepasst.

Weiß nicht was diese aussage soll.. nur weil der code vorher nicht schön war ist ja kein freifahrtsschein es noch beliebig weiter zu treiben und ihn ggf. Am ende sogar komplett unwartbar zu hinterlassen..

Reply:
Zu der Klasse für die Rechteabfragen habe ich weiter oben auch eine Idee notiert: https://github.com/redaxo/redaxo/pull/3667#discussion_r499235376

Also da ist mein Vorschlag, alle Rechtemethoden bzgl Mediacats in `rex::getUser()->getComplexPerm('media')->...` zu packen.